core/app/models/spree/variant.rb

```diff
@@ -228,41 +220,45 @@ def dimension
 
     private
 
-      def set_master_out_of_stock
-        if product.master && product.master.in_stock?
-          product.master.stock_items.update_all(:backorderable => false)
-          product.master.stock_items.each { |item| item.reduce_count_on_hand_to_zero }
-        end
-      end
+    def quantifier
+      Spree::Stock::Quantifier.new(self)
+    end
 
-      # Ensures a new variant takes the product master price when price is not supplied
-      def check_price
-        if price.nil? && Spree::Config[:require_master_price]
-          raise 'No master variant found to infer price' unless (product && product.master)
-          raise 'Must supply price for variant or master.price for product.' if self == product.master
-          self.price = product.master.price
-        end
-        if currency.nil?
-          self.currency = Spree::Config[:currency]
-        end
+    def set_master_out_of_stock
+      if product.master && product.master.in_stock?
+        product.master.stock_items.update_all(:backorderable => false)
+        product.master.stock_items.each { |item| item.reduce_count_on_hand_to_zero }
```

Comment:
Pass `&:reduce_count_on_hand_to_zero` as an argument to `each` instead of a block.